maven-core/src/main/java/org/apache/maven/project/DefaultProjectDependenciesResolver.java

```diff
@@ -160,7 +160,9 @@ public DependencyResolutionResult resolve(DependencyResolutionRequest request)
             result.setCollectionErrors(e.getResult().getExceptions());
 
             throw new DependencyResolutionException(
-                    result, "Could not resolve dependencies for project " + project.getId() + ": " + e.getMessage(), e);
+                    result,
+                    "Could not collect dependencies for project " + project.getId(),
+                    logger.isDebugEnabled() ? e : null);
```

Comment:
Why omitting the exception?

Reply:
When we have exception in stack all message from all exception will be added to output in normal run we will have some information duplicate now

Reply:
But isn't that a problem of the catcher, not the thrower?

Reply:
If we have more detailed info at debug level, especially in case of an error, it may make sense to indicate so in the non detailed error message ?

Reply:
detailed contains more info which is not available in `o.e.aether.resolution.ArtifactResolutionException`
`ArtifactResolutionException.getMessage` returns error for last repository

Reply:
This is one of the reasons led me to DOXIASITETOOLS-303....

Reply:
I still have trouble understanding that. Maybe the caller code needs to change or the message rendered by this exception needs to be improved...

Reply:
without passing exception (current proposition) we will have:
```
[ERROR] Failed to execute goal on project test: Could not resolve dependencies for project org.apache.maven.its.mng3477:test:jar:1
[ERROR] dependency: org.apache.maven.its.mng3477:dep:jar:1.0 (compile)
[ERROR] 	Could not find artifact org.apache.maven.its.mng3477:dep:jar:1.0 in maven-core-it (http://localhost:64468/repo)
[ERROR] 	Could not find artifact org.apache.maven.its.mng3477:dep:jar:1.0 in central (http://localhost:64468/repo)
[ERROR] -> [Help 1]
```

with exception will be as:

```
[ERROR] Failed to execute goal on project test: Could not resolve dependencies for project org.apache.maven.its.mng3477:test:jar:1
[ERROR] dependency: org.apache.maven.its.mng3477:dep:jar:1.0 (compile)
[ERROR] 	Could not find artifact org.apache.maven.its.mng3477:dep:jar:1.0 in central (http://localhost:64398/repo)
[ERROR] 	Could not find artifact org.apache.maven.its.mng3477:dep:jar:1.0 in maven-core-it (http://localhost:64398/repo): The following artifacts could not be resolved: org.apache.maven.its.mng3477:dep:jar:1.0 (absent): Could not find artifact org.apache.maven.its.mng3477:dep:jar:1.0 in central (http://localhost:64398/repo)
[ERROR] -> [Help 1]
```

so it is a reason

Reply:
I hope that simply message will be more readable for user, if user need more details for debug or investigation can execute in verbose/debug mode

Reply:
VERY UGLY. I guess this needs a long term solution.

Reply:
we can think about it ... how to reporting a problems to user but it is another issue for me, now I did as simple as possible 
we also have many other UGLY code ... 😸